### DISCUSSION TITLE
What is the purpose of setTimeout?

### DISCUSSION THREAD

**Discussion by niutech** [Kalabasa/htmz#16] [Q&A] [answered]
You didn't explain it in How it works.

**Comment by jashan12002** [comment]
**setTimeout** is used when we want to run a function after a set of time is passed.

**Reply by niutech** [reply]
I know what setTimeout does, but there is no explicit timeout being set, so what is its purpose in this particular case?

**Comment by Kalabasa** [comment, marked as answer]
I've just added an explanation to setTimeout in the dev snippet! 

tl;dr: The browser automatically scrolls DOM changes into view in response to user interaction.

See 1a1e82e

**Reply by Kalabasa** [reply]
This is interesting: If your webapp is a fullscreen SPA (no scrolling), then you can remove the setTimeout wrapper!

Which results in a **150-byte snippet**, which is very tempting... But it makes tabbed UIs annoying (such as the demo section in the htmz website, so would not be a good impression).

**Reply by Malix-Labs** [reply]
> https://github.com/Kalabasa/htmz/blob/1a1e82e15ec25f55ef49740a6b9aada5ba98aeb3/htmz.dev.html#L5-L7

**Reply by Malix-Labs** [reply]
- [Comment](https://github.com/Kalabasa/htmz/discussions/16#discussioncomment-8576115) from @Kalabasa:

  > [⋯](https://github.com/Kalabasa/htmz/discussions/16#discussioncomment-8576115) If your webapp is a fullscreen SPA (no scrolling), then you can remove the setTimeout wrapper [⋯](https://github.com/Kalabasa/htmz/discussions/16#discussioncomment-8576115)

  Wouldn't it thus break responsiveness?
  In that case, it is interesting to know, but for sure not desired unless unresponsiveness is not an issue and saving 16 bytes is critical.
